MIGRATING_FROM_V1.md

```diff
@@ -0,0 +1,262 @@
+# Migration guide from Trampoline V1 script
+
+This guide is solely for migrating from Trampoline V1 script. We
+recommend that you read README.md before diving into this guide.
+
+In this guide, we'll use
+[getting-started-python](https://github.com/GoogleCloudPlatform/getting-started-python)
+as an example repository.
+
+## The first step - local run
+
+The first step of the migration is to run the tests locally with
+Trampoline V2 script. Copy trampoline_v2.sh into your repository. A
+natural choice is to place the script alongside with the
+`trampoline.sh`.
+
+Here is an exmaple with our example repo. Make sure it has an
+executable bit.
+
+```sh
+$ git clone git@github.com:GoogleCloudPlatform/getting-started-python.git
+$ cd getting-started-python
+$ cp ../docker-ci-helper/trampoline_v2.sh .kokoro
+$ ls -al .kokoro/trampoline_v2.sh
+-rwxr-x--- 1 user group 14591 Jul  9 20:28 .kokoro/trampoline_v2.sh
+```
+
+Then you will need to set some environment variables. We can mostly
+reuse the current settings modulo small changes. Note we can omit the
+`github/getting-started-python` part from `TRAMPOLINE_BUILD_FILE`. In
+our example repo, you can see the current settings in
+`.kokoro/common.cfg`.
+
+Here we set mandatory environment variables `TRAMPOLINE_IMAGE` and
+`TRAMPOLINE_BUILD_FILE`, as well as optional `TRAMPOLINE_VERBOSE` for
+easier debugging.
+
+```sh
+export TRAMPOLINE_IMAGE=gcr.io/cloud-devrel-kokoro-resources/python@sha256:4b6ba8c199e96248980db4538065cddeea594138b9b9fb2d0388603922087747
+export TRAMPOLINE_BUILD_FILE=.kokoro/system_tests.sh
+export TRAMPOLINE_VERBOSE=true
+```
+
+Download the docker image locally.
+
+```sh
+docker pull "${TRAMPOLINE_IMAGE}"
+```
+
+Then run the script.
+
+```sh
+.kokoro/trampoline_v2.sh
+```
+
+The first run will likely fail, but make sure to see outputs like this:
+
+```
+2020-07-09T20:40:24Z: Running the tests in a Docker container.
+================================================================
+docker run ...
+```
+
+At least, it successfully run the container. The test likely fails due
+to some differences between Trampoline V1 and V2. We'll fix the build
+script in the next section.
+
+## Modify the build script
+
+Our first run failed with the following message:
+
+```
+.kokoro/system_tests.sh: line 21: cd: github/getting-started-python: No such file or directory
+```
+
+We can modify this line:
```

Comment:
Perhaps link to a permalinked version of the file at this commit?

Reply:
Sure, done